src/libexpr/eval.hh

```diff
@@ -360,6 +361,10 @@ private:
 
 public:
 
+    std::vector<PosIdx> stack = {};
```

Comment:
It'd be good to have primops as well. They all have an internal name besides their `builtins` attribute, so this can work:

```suggestion
    std::vector<std::variant<PosIdx, std::string>> stack = {};
```

Reply:
Good idea.

Reply:
Perhaps a more general solution is to allow primops to have their own type of `Pos`, but that may or may not be complicated because of `Pos`'s other uses, so I'd be fine with either solution.